### PR TITLE
segacd.xml, megacdj.xml: New software list items added

### PR DESCRIPTION
**Pull request by FoxhackDN**
New working software list items
-------------------------------
segacd.xml:
Compton's Interactive Encyclopedia v2.00S (USA) [redump.org] Note! Color Mechanica (USA) [redump.org]
Note! Color Mechanica (USA, alt) [redump.org]
What is X'Eye Multi Entertainment System (USA) [redump.org]

megacdj.xml:
Heavenly Symphony - Formula One World Championship 1993 Hibaihin (Japan) [redump.org]
Keiou Yuugekitai Taikenban Hibaihin (Japan) [redump.org]
Lunar - Eternal Blue Hibaihin Auto Demo (Japan) [redump.org]
Microcosm Demo CD (Japan) [redump.org]
Night Trap Hibaihin (Japan) [redump.org]
Popful Mail Taikenban Hibaihin (Japan) [redump.org]
Silpheed Hibaihin (Japan) (Fixed) [redump.org]
Sonic The Hedgehog CD Hibaihin (Japan) [redump.org]
Thunderhawk Hibaihin (Japan) [redump.org]
Urusei Yatsura - Dear My Friends Hibaihin (Japan) [redump.org]
Yumemi Yakata no Monogatari Hibaihin (Japan) [redump.org]
WonderMega Collection - Game Garden (Japan, alt) [redump.org]

New software list items marked not working
------------------------------------------
segacd.xml:
Surgical Strike (Brazil, 32X) [redump.org]

megacdj.xml:
Psychic Detective Series vol.3 - AYA Auto Demo (Japan) [redump.org]
Silpheed Hibaihin (Japan) [redump.org]

### CODE REVIEW

**Comment by angelosa** [comment]
fwiw I'm not sure if I want demos as a `cloneof` a retail disc. I'm not even sure if I want demos to coexist in the same `.xml` for QA regtesting and software grouping reasons.

File: hash/megacdj.xml [hunk 1]
```diff
@@ -5157,6 +5157,92 @@ Black screen, requires sub CPU to be overclocked by 1.5x
...
+		<rom name="Heavenly Symphony - Formula One World Championship 1993 (Japan) (Demo) (Track 35).bin" size="16186464" crc="950b6a34" />
+		<rom name="Heavenly Symphony - Formula One World Championship 1993 (Japan) (Demo) (Track 36).bin" size="14866992" crc="108f5a17" />
+		<rom name="Heavenly Symphony - Formula One World Championship 1993 (Japan) (Demo) (Track 37).bin" size="43267392" crc="e55cb55b" />
+		-->
+		<description>Heavenly Symphony - Formula One World Championship 1993 Hibaihin (Japan)</description>
+		<year>1994</year>
+		<publisher>Sega</publisher>
+		<info name="serial" value="SG-6034"/>
+		<info name="disk_ring" value="SG-6034-R1M V"/>
+		<info name="alt_title" value="ヘブンリー シンフォニー フォームラワン ワールド チャンピオンシップ 1993 非売品"/>
```
Comment (cuavas) [inline hash/megacdj.xml:5206]: Why does this software list use `disk_ring` info elements when megacd.xml uses `ring_code`?  It’s not even consistent between the two Mega CD lists.  Also, which code are these supposed to represent?  Mastering code?  Matrix?  Mastering SID?  Mould SID?  A specific info element should be used for each (e.g. cdi.xml uses `cd_matrix` for matrix codes).

Comment (FoxhackDN) [reply hash/megacdj.xml:5206]: Oh, sorry, I didn't know there were several options for metadata. My `neocd,xml` submission will also need correcting. Is there a list somewhere that I can check, for reference, or did you just mention them all?

I'll fix all of these later today, sorry.

Comment (cuavas) [reply hash/megacdj.xml:5206]: No need to be sorry, these three lists were a mess before you got to them.

There’s a little info on the different “ring codes” here: https://www.rushmediaprint.com/cd-matrix-code-and-sid-codes/

I suspect all the codes in the three Mega CD software lists are matrix codes.  I’d suggest changing all three to use `cd_matrix` to match cdi.xml.

**Comment by cuavas** [comment]
> fwiw I'm not sure if I want demos as a `cloneof` a retail disc. I'm not even sure if I want demos to coexist in the same `.xml` for QA regtesting and software grouping reasons.

There’s no reason to put them in a separate software list, that just makes life hard with extra clutter.  I can see the argument for not marking them as clones – they’re definitely not a substitute.

**Comment by FoxhackDN** [comment]
> > fwiw I'm not sure if I want demos as a `cloneof` a retail disc. I'm not even sure if I want demos to coexist in the same `.xml` for QA regtesting and software grouping reasons.
> 
> There’s no reason to put them in a separate software list, that just makes life hard with extra clutter. I can see the argument for not marking them as clones – they’re definitely not a substitute.

I agree with both of you. Most of these are earlier builds of the full game, but Lunar, Urusei Yatsura and Yumemi Yakata no Monogatari are autoplaying demos that don't actually receive any input from the player. (Silpheed is a mix of demo and autoplaying content.)

I'd be fine with removing the clone relationship from those three games.

File: hash/megacdj.xml (hunk 1 above)
Comment (FoxhackDN) [reply hash/megacdj.xml]: Allright, done. I also removed an extra tag I left in by accident.

**Pull request merged**